pkg/logql/syntax/ast.go

```diff
@@ -956,31 +967,27 @@ func (e *LabelFmtExpr) String() string {
 	return sb.String()
 }
 
-type JSONExpressionParser struct {
+type JSONExpressionParserExpr struct {
```

Comment:
All structs that implement the `Expr` interface and are part of the parser tree (AST) have the suffix `Expr`.